tools/aiecc/aiecc/main.py

```diff
@@ -273,15 +273,36 @@ async def process_host_cgen(self):
           cmd += ['--gcc-toolchain=%s/usr' % opts.sysroot]
 
       thispath = os.path.dirname(os.path.realpath(__file__))
-      runtime_xaiengine_path = os.path.join(thispath, '..','..','runtime_lib', opts.host_target.split('-')[0], 'xaiengine')
-      xaiengine_include_path = os.path.join(runtime_xaiengine_path, "include")
-      xaiengine_lib_path = os.path.join(runtime_xaiengine_path, "lib")
+
+      # We can have differnet libxaie targets and need to point to the proper one
+      if (opts.host_target.split('-')[0] == 'x86_64'):
+        runtime_xaiengine_path = os.path.join(aiecc.configure.libxaie_x86_dir)
+        xaiengine_include_path = os.path.join(runtime_xaiengine_path, "include")
+        xaiengine_lib_path = os.path.join(runtime_xaiengine_path, "lib")
+      else:
+        runtime_xaiengine_path = os.path.join(thispath, '..','..','runtime_lib', opts.host_target.split('-')[0], 'xaiengine')
+        xaiengine_include_path = os.path.join(runtime_xaiengine_path, "include")
+        xaiengine_lib_path = os.path.join(runtime_xaiengine_path, "lib")
+      
+      # Adding the runtime test library, which will always be in the same spot
       runtime_testlib_path = os.path.join(thispath, '..','..','runtime_lib', opts.host_target.split('-')[0], 'test_lib', 'lib')
-      memory_allocator = os.path.join(runtime_testlib_path, 'libmemory_allocator_ion.a')
+
+      if (opts.host_target.split('-')[0] == 'x86_64'):
+        memory_allocator = os.path.join(runtime_testlib_path, 'libmemory_allocator_air.a')
+      else: 
+        memory_allocator = os.path.join(runtime_testlib_path, 'libmemory_allocator_ion.a')
 
       cmd += [memory_allocator]
       cmd += ['-I%s' % xaiengine_include_path]
       cmd += ['-L%s' % xaiengine_lib_path]
+      cmd += ["-Wl,-R%s" % xaiengine_lib_path]
+
+      if(opts.host_target.split('-')[0] == 'x86_64'):
+        mlir_air_include_path = os.path.join(aiecc.configure.mlir_air_dir, "build-pcie", "runtime_lib", "x86_64", "airhost", "include");
+        mlir_air_lib_path = os.path.join(aiecc.configure.mlir_air_dir, "build-pcie", "runtime_lib", "x86_64", "airhost");
```

Comment:
can't get checked in like this.   let's bring this into mlir-aie

Reply:
This dependency with AIR has been removed.